cdap-app-fabric/src/main/java/io/cdap/cdap/internal/bootstrap/BootstrapService.java

```diff
@@ -82,7 +84,7 @@ protected void startUp() {
       } catch (InterruptedException e) {
         LOG.info("Bootstrapping could not complete due to interruption. It will be re-run the next time CDAP starts.");
       }
-    });
+    }).get();
```

Comment:
it's better to keep this asynchronous. Bootstrapping can take a while, and we don't want to hold up startup for unrelated services.